### ISSUE TITLE
Abort seems not working properly

### ISSUE THREAD

**Issue by intelcoder** [forwardemail/superagent#1562]
I am currently using superagent with saga and like to keep using it

here is some code I have been trying. somehow promise.abort is undefined even though I can see a lot of request is on pending on the dev tool.

When I tried to pass promise to the redux and abort in there. It only cancel first pending request somehow.
```
 yield put(getSearchActions.request({ payloadJson }))
    if(fetching && promise && promise.abort) {
      console.log(' I am aborting')
      promise.abort()
    }
    promise = apiSearch({ data: payload })
    const result = yield promise

    yield put(getSearchActions.success(payload, result.response))
    promise = null
    return null
```

However, I tried with axios. this code actually cancelling. With axios I found out I need to generate cancelToken after I cancel. Do I have to do the similar thing with superagent?

 ```
if(fetching) {
      source.cancel()
      source = CancelToken.source()
    }
    yield put(getSearchActions.request({ payloadJson }))
    yield delay(200)
    try {
      const result = yield instance.get('/url', {
        params: payload,
        cancelToken: source.token,
      })
      yield put(getSearchActions.success(payload, { data: result.data }))
    } catch(e) {
      console.log(e)
      yield put(getSearchActions.failure(payload, null))
    }
```

**Comment by niftylettuce** [comment]
PR welcome, it would be great to first add a failing test, and then the fix if you don't mind

**Comment by titanism** [comment]
Keep the SuperAgent Request returned directly and call request.abort(); do not replace it with .then()’s native Promise or an async wrapper. Handle expected ABORTED rejection.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.

**Comment by titanism** [comment]
Closing this for now. If this still affects v10.4.0, please open a new issue with a current reproduction.

v10.4.0 is available on npm: https://www.npmjs.com/package/superagent/v/10.4.0
GitHub release: https://github.com/forwardemail/superagent/releases/tag/v10.4.0

Consider supporting our efforts to maintain this project by using Forward Email at https://forwardemail.net.  We run the only 100% open source and 3rd party audited email service in the world.
